kvserver: plumb priority to process

wenyihu6 opened the pull request.

wenyihu6 committed:
add priority changes

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152319)
<!-- Reviewable:end -->

wenyihu6 committed:
remove priority setting

wenyihu6 committed:
plumb priority for purgatory queue

wenyihu6 committed:
add obs

wenyihu6 committed:
requeue

wenyihu6 committed:
pri

wenyihu6 committed:
allocator: move isDecommissionAction to allocatorimpl

This commit refactors isDecommissionAction into allocatorimpl for consistency
with other similar helpers like allocatorActions.{Add,Replace,Remove}. This
change has no behavior changes but to make future commits easier.

wenyihu6 committed:
priority inversion

wenyihu6 committed:
kvserver: fix comment when dropping due to exceeding size

Previously, the comment on the queue incorrectly stated that it removes the
lowest-priority element when exceeding its maximum size. This was misleading
because heap only guarantees that the root is the highest priority, not that
elements are globally ordered. This commit updates the comment to clarify that
the removed element might not be the lowest priority. Ideally, we should drop
the lowest priority element when exceeding queue size, but heap doesn't make
this very easy.

wenyihu6 committed:
print top k

wenyihu6 committed:
kvserver: simplify logging in maybeEnqueueProblemRange

This commit simplifies the logging in `maybeEnqueueProblemRange` to log two
booleans directly.

wenyihu6 committed:
enqueue

wenyihu6 committed:
enqueue at

wenyihu6 committed:
Vprint top k ranges

wenyihu6 committed:
add to purgatory

wenyihu6 committed:
knobs

# Conflicts:
#	pkg/kv/kvserver/queue.go
#	pkg/kv/kvserver/replicate_queue_test.go

wenyihu6 committed:
knobs

wenyihu6 committed:
add metrics

wenyihu6 committed:
tests

wenyihu6 committed:
firm up the tests

wenyihu6 committed:
kvserver: add logging for ranges dropped from base queue

This commit adds logging for ranges dropped from the base queue due to exceeding
max size, improving observability. The log is gated behind V(1) to avoid
verbosity on nodes with many ranges.
